internal/datadog/agent.go

```diff
@@ -132,7 +132,7 @@ func (p *traceagent) goDrain() {
 
 // Ingest processes the given spans within the traceagent and outputs stats through the output channel
 // provided to newAgent. Do not call Ingest on an unstarted or stopped traceagent.
-func (p *traceagent) Ingest(ctx context.Context, traces ptrace.Traces) {
+func (p *Traceagent) Ingest(ctx context.Context, traces ptrace.Traces) {
```

Comment:
nit: it would make sense to capitalise the "a" too (TraceAgent), if you did the the "T"